### ISSUE TITLE
It's weird that it says "missing field(s):" and no fields

### ISSUE THREAD

**Issue by andrewballantyne** [tektoncd/pipeline#2235]
# Expected Behavior
To see what fields are required of me (when omitting content from `spec`).

# Actual Behavior

```
$ oc apply -f no-task-pipeline.yaml
Error from server (BadRequest): error when creating "no-task-pipeline.yaml": admission webhook "webhook.tekton.dev" denied the request: mutation failed: missing field(s):
```

# Steps to Reproduce the Problem

1. Apply an empty `spec` Pipeline to the OpenShift Pipelines Operator

# Additional Info

- Kubernetes version:

  **Output of `kubectl version`:**

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.3", GitCommit:"2d3c76f9091b6bec110a5e63777c332469e0cba2", GitTreeState:"clean", BuildDate:"2019-08-19T12:36:28Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17+", GitVersion:"v1.17.1", GitCommit:"f17a1cc", GitTreeState:"clean", BuildDate:"2020-03-15T08:08:33Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
```

- Tekton Pipeline version:

  **Output of `tkn version` or `kubectl get pods -n tekton-pipelines -l app=tekton-pipelines-controller -o=jsonpath='{.items[0].metadata.labels.version}'`**

```
$ tkn version
Client version: 0.8.0
Pipeline version: unknown
```

The file I used:
```
$ cat no-task-pipeline.yaml
apiVersion: tekton.dev/v1alpha1
kind: Pipeline
metadata:
  name: no-task-pipeline
spec:
```

Also tried it in the OpenShift Console UI (with & without `tasks:`)
![image](https://user-images.githubusercontent.com/8126518/76750800-3b1f2780-6755-11ea-9eb3-41b7859b73ac.png)

**Comment by GregDritschler** [comment]
I can fix this one (and also #1844 which is the same issue).

**Comment by GregDritschler** [comment]
The "missing field(s)" error is coming from pipeline_validation.go.

```
func (ps *PipelineSpec) Validate(ctx context.Context) *apis.FieldError {
	if equality.Semantic.DeepEqual(ps, &PipelineSpec{}) {
		return apis.ErrMissingField(apis.CurrentField)
	}
```

apis.ErrMissingField and apis.CurrentField are defined in vendor/knative.dev/pkg/apis/field_error.go.

apis.CurrentField is a constant empty string.  It is intended to be used to say that the path where the
error was found is defined by another method such as apis.ViaField.  That doesn't happen here so the path remains empty in the final message.

This "missing fields(s)" check in pipeline validation is debatable.  According to the pipeline documentation all of the fields under "spec" are optional.  This means an empty "spec" is valid and there shouldn't be an error.  On the other hand there is value in trying to catch simple mistakes
such as a misaligned "spec" keyword.

I am going to attach a PR with my attempt to resolve it.
